src/test/java/org/opentripplanner/middleware/controllers/api/TrackedTripControllerTest.java

```diff
@@ -504,6 +516,111 @@ void canNotUpdateCompletedJourney() throws Exception {
+        ManageTripTracking.otpResponseProviderOverride = mockOtpResponse;
+        var deviatedPosition = new TrackingLocation(Instant.now(), 33.94412, -83.98899);
+        var reroutingPoint = new Coordinates(expectedReroutedItinerary.legs.get(0).steps.get(2));
+        var reroutingPointPosition = new TrackingLocation(Instant.now(), reroutingPoint.lat,reroutingPoint.lon);
+
+        // Start tracking.
+        var response = makeRequest(
+            START_TRACKING_TRIP_PATH,
+            JsonUtils.toJson(startTrackingPayload),
+            headers,
+            HttpMethod.POST
+        );
+
+        var startTrackingResponse = JsonUtils.getPOJOFromJSON(response.responseBody, TrackingResponse.class);
+        trackedJourney = Persistence.trackedJourneys.getById(startTrackingResponse.journeyId);
+        assertEquals(HttpStatus.OK_200, response.status);
+
+        // Update tracking from a 'deviated' position.
+        response = makeRequest(
+            UPDATE_TRACKING_TRIP_PATH,
+            JsonUtils.toJson(createUpdateTrackingPayload(
+                trackedJourney.id,
+                List.of(deviatedPosition)
+            )),
+            headers,
+            HttpMethod.POST
+        );
+
+        // Confirm traveler is classed as 'deviated'.
+        var updateTrackingResponse = JsonUtils.getPOJOFromJSON(response.responseBody, TrackingResponse.class);
+        assertEquals(TripStatus.DEVIATED.name(), updateTrackingResponse.tripStatus);
+
+        // Reroute trip from 'deviated' position.
+        var reroutedItinerary = ManageTripTracking.rerouteTrip(
+            new TripTrackingData(rerouteMonitoredTrip, trackedJourney, List.of(deviatedPosition))
+        );
+        assertEquals(expectedReroutedItinerary.duration, reroutedItinerary.duration);
+        TrackedJourney updated = Persistence.trackedJourneys.getById(trackedJourney.id);
+        assertEquals(1, updated.reroutings.size());
+        assertEquals(trackedJourney.longestConsecutiveDeviatedPoints, updated.longestConsecutiveDeviatedPoints);
+
+        MonitoredTrip trip = Persistence.monitoredTrips.getById(rerouteMonitoredTrip.id);
+        assertEquals(expectedReroutedItinerary.duration, trip.journeyState.matchingItinerary.duration);
+        assertEquals(expectedReroutedItinerary.legs.size(), trip.journeyState.matchingItinerary.legs.size());
+
+        // Update tracking from start of rerouted position.
+        response = makeRequest(
+            UPDATE_TRACKING_TRIP_PATH,
+            JsonUtils.toJson(createUpdateTrackingPayload(
+                trackedJourney.id,
+                List.of(deviatedPosition)
+            )),
+            headers,
+            HttpMethod.POST
+        );
+
+        // Confirm traveler is no longer 'deviated'.
+        updateTrackingResponse = JsonUtils.getPOJOFromJSON(response.responseBody, TrackingResponse.class);
+        assertNotEquals(TripStatus.DEVIATED.name(), updateTrackingResponse.tripStatus);
+
```

Comment:
Try to insert a call to `CheckMonitoredTrip` and assert that it doesn't revert the matching itinerary to what was before.

Reply:
I'm not 100% sure on this. I've provided an updated for review: https://github.com/ibi-group/otp-middleware/pull/286/commits/23d04728e3e4d229384d0e37b6ca821b9f416208

Reply:
Nice, I think that will work.